### DISCUSSION TITLE
No connection to display???

### DISCUSSION THREAD

**Discussion by jazzmonger** [HASwitchPlate/HASPone#222] [Q&A] [answered]
**EDIT**: I changed the baud rate to 115k and it started working....  is this the correct baud rate?
EDIT2: I spoke too soon... its hit or miss...  sometimes it connects, mostly it doesn't.  I tried a different ESP w/ the same results.  swapping TX/RX no effect.

Go the D1 Mini flashed & configured.  It all looks like its working, but the display just boots and goes into its demo mode.

The log shows a connection to the display ( I think ), but logs of "invalid" errors...  I'm all ears on what to try.  How do you get the display out of demo mode????

```
[+0.175s] HMI: Waiting for LCD connection
[+2.176s] HMI: No Nextion response, attempting to set serial speed to 115200
[+2.183s] HMI: Sending bauds=115200 @2400 baud
[+2.287s] HMI: Sending bauds=115200 @4800 baud
[+2.346s] HMI: Sending bauds=115200 @9600 baud
[+2.377s] HMI: Sending bauds=115200 @19200 baud
[+2.395s] HMI: Sending bauds=115200 @31250 baud
[+2.408s] HMI: Sending bauds=115200 @38400 baud
[+2.419s] HMI: Sending bauds=115200 @57600 baud
[+2.428s] HMI: Sending bauds=115200 @115200 baud
[+2.435s] HMI: Sending bauds=115200 @230400 baud
[+2.441s] HMI: Sending bauds=115200 @250000 baud
[+2.446s] HMI: Sending bauds=115200 @256000 baud
[+2.452s] HMI: Sending bauds=115200 @512000 baud
[+2.457s] HMI: Sending bauds=115200 @921600 baud
[+2.462s] HMI: Waiting again for LCD connection
[+4.463s] HMI: LCD connection timed out
[+4.466s] HMI: LCD not responding, continuing program load
[+4.475s] HMI OUT: 'p[0].b[1].font=6'
```

**Comment by aderusha** [comment]
Sounds like you might not have flashed [the TFT file]([HASwitchPlate.tft](https://github.com/HASwitchPlate/HASPone/blob/main/Nextion_HMI/HASwitchPlate.tft)) to the display.  Write that file to an SD card and boot the Nextion with that inserted.

**Comment by jazzmonger** [comment]
I flashed the D1 mini.  I didnt see anything about a flash card.


I have an SD card, write what "file"??

**Comment by aderusha** [comment]
Thought I linked that firmware but apparently not which makes for some confusion!  Sorry man, here's the link: https://github.com/HASwitchPlate/HASPone/blob/main/Nextion_HMI/HASwitchPlate.tft

The HASPone project has firmware for the ESP8266 and firmware for the Nextion.  Nextion firmware comes as a .TFT file, so download that link above, put it on an SD card, unplug your Nextion, put the SD card in, then plug the Nextion back in to power it up.  It should recognize your firmware and start an upload process automatically.  Afterwards, remove SD card, power down, then power back up.

**Reply by jazzmonger** [reply]
I format an SD card and put your "HASwitchPlate.tft" file on it.

upon boot up I get " SD card Update... The system detects multiple tft files and can not be upgraded.  please remove the extra tft file"

do I need to rename it to something else?
I tried it on 2 display units - same msg.

EDIT:  I used MACOS to write the file.  rewriting w/ Linux solved that.  but its now stuck on "SD Card Update"

**Comment by jazzmonger** [comment, marked as answer]
so... I had to use BlueHarvest macos app to prevent my mac from writing its stupid hidden files.  http://www.zeroonetwenty.com/blueharvest/
Also, these displays only support SD cards <=32GB.  You know how many devices in my house I had to look at to find one of those dinosaurs?
once those dragons were slayed, the firmware on the display upgraded.

It would be nice if this project gave a little back ground so newbies like me can figure this stuff out a little easier.

**The Nextion display has its own internal RTOS  and it needs a .TFT file to be loaded via the SD card run the display. Unlike other "dumb" displays, that file specifies all kinds of things to make these displays work.  it also only supports SD Card updating .tft files on 32gb SD cards written by linux or windows using FAT32 (not MACOS)**

see how easy that was?  if this was spelled out, it would have saved me a day of screwing around. The positive note is I now fully get what you are doing w/ this project.  I also had to do summersaults on my mac running Parallels to get the Nextion Editor to run... 
And the display works GREAT!!

I tried to help you out and add this edit to your docs, but you don't allow edits...

**Comment by aderusha** [comment]
Great news man, glad you got it sorted!
